### ISSUE TITLE
Model speed

### ISSUE THREAD

**Issue by fanaev** [HamadYA/GhostFaceNets#4]
Hello! I m trying to create cpu-real-time face recognition system based on GhostFaceNets. Detection model that i use shows 50-60 fps. But when i add verification step (Feature extraction via GhostFaceNets) the speed of system drops to 3 fps (checked on Macbook Pro A1708 and Magicbook15). is that how it should be? Is it any option to increase speed? Have you tried to create cpu-real-time face recognition system based on GhostFaceNets? What fps did you gained?
I've used code below
`feauture_extractor = keras.models.load_model('models/GN_W0.5_S2_ArcFace_epoch16 (1).h5', compile=False)`
`for faces in faces:`
`   ...`
`   cropped_face = img[y:y + h, x:x + w]`
`   cropped_face = tf.image.resize(cropped_face, [112, 112]) /255.0`
`   cropped_face = tf.expand_dims(cropped_face, axis=0)`
`   face_embedding = feauture_extractor(cropped_face)`

**Comment by HamadYA** [comment]
Hi,

Let me test it and get back to you by the end of the day hopefully.

**Comment by HamadYA** [comment]
> Hello! I m trying to create cpu-real-time face recognition system based on GhostFaceNets. Detection model that i use shows 50-60 fps. But when i add verification step (Feature extraction via GhostFaceNets) the speed of system drops to 3 fps (checked on Macbook Pro A1708 and Magicbook15). is that how it should be? Is it any option to increase speed? Have you tried to create cpu-real-time face recognition system based on GhostFaceNets? What fps did you gained? I've used code below `feauture_extractor = keras.models.load_model('models/GN_W0.5_S2_ArcFace_epoch16 (1).h5', compile=False)` `for faces in faces:` ` ...` ` cropped_face = img[y:y + h, x:x + w]` ` cropped_face = tf.image.resize(cropped_face, [112, 112]) /255.0` ` cropped_face = tf.expand_dims(cropped_face, axis=0)` ` face_embedding = feauture_extractor(cropped_face)`

Hi,

Let me test it and get back to you by the end of the day hopefully.

**Comment by HamadYA** [comment]
> GN_W0.5_S2_ArcFace_epoch16 (1).h5

I did multiple tests on CPU (in the attached zip), the inference speed is pretty high. Perhaps you have something that keeps the model delayed. Please check and make sure that this slow fps is caused by the GhostFaceNet, if so, try to use TensorFlowLite to compress the model.

Let me know if you have any progress or the slow fps still resist. I will develop a face detector and use GhostFaceNet to test on CPU meanwhile. Will send the results tomorrow hopefully.

[inference_test.zip](https://github.com/HamadYA/GhostFaceNets/files/11444093/inference_test.zip)

**Comment by fanaev** [comment]
Thank you very much, i will try it

**Comment by fanaev** [comment]
i found that this model [GhostFaceNetV1-0.5-2 (A)](https://github.com/HamadYA/GhostFaceNets/releases/download/v1.0/GN_W0.5_S2_ArcFace_epoch16.h5) (which i used firstly) is slower than [GhostFaceNetV1-1.3-2 (A)](https://github.com/HamadYA/GhostFaceNets/releases/download/v1.3/GhostFaceNet_W1.3_S2_ArcFace.h5). 
**GhostFaceNetV1-0.5-2 inference time**:
![image](https://github.com/HamadYA/GhostFaceNets/assets/81157248/7dee561d-4f05-42ed-a2dc-52a37c857ec8)
**GhostFaceNetV1-1.3-2 inference time:**
![image](https://github.com/HamadYA/GhostFaceNets/assets/81157248/b5678c8f-b73a-4d01-ad1e-da6bc3871318)




However GhostFaceNetV1-0.5-2 have less parameters than GhostFaceNetV1-1.3-2 according to paper:
<img width="331" alt="image" src="https://github.com/HamadYA/GhostFaceNets/assets/81157248/3b3af0bc-23da-4938-986d-d6dcdd41e8f3">

**Comment by HamadYA** [comment]
> i found that this model [GhostFaceNetV1-0.5-2 (A)](https://github.com/HamadYA/GhostFaceNets/releases/download/v1.0/GN_W0.5_S2_ArcFace_epoch16.h5) (which i used firstly) is slower than [GhostFaceNetV1-1.3-2 (A)](https://github.com/HamadYA/GhostFaceNets/releases/download/v1.3/GhostFaceNet_W1.3_S2_ArcFace.h5). **GhostFaceNetV1-0.5-2 inference time**: ![image](https://user-images.githubusercontent.com/81157248/237750228-7dee561d-4f05-42ed-a2dc-52a37c857ec8.png) **GhostFaceNetV1-1.3-2 inference time:** ![image](https://user-images.githubusercontent.com/81157248/237750467-b5678c8f-b73a-4d01-ad1e-da6bc3871318.png)
> 
> However GhostFaceNetV1-0.5-2 have less parameters than GhostFaceNetV1-1.3-2 according to paper: <img alt="image" width="331" src="https://user-images.githubusercontent.com/81157248/237749776-3b3af0bc-23da-4938-986d-d6dcdd41e8f3.png">

Hi,
Sorry for being late in replying, I am in an exams period in my university.

This should not be the case, based on the experiments done to produce the results in the paper, GhostFaceNets with width of 0.5 were extremely fast to train and test.

I will develop a full system for you on CPU as soon as possible; but again, currently busy with the exams so I cannot promise when it will be ready. Hopefully in one day time.

**Comment by fanaev** [comment]
>I will develop a full system for you on CPU as soon as possible; 

Don't bother yourself with it. My CPU is just so slow to use this model in real-time system.
